vsphere_plugin_common/tests/plugin_common_test.py

```diff
@@ -23,7 +30,85 @@
 import vsphere_plugin_common
 
 
+class WebServer(object):
+    def __init__(self, port=4443,
+                 key='private.pem', cert='public.pem',
+                 badkey='badkey.pem', badcert='badcert.pem'):
+        self.key = key
+        self.cert = cert
+        self.badkey = badkey
+        self.badcert = badcert
+        for i in range(0, 6):
+            try:
+                self.httpd = BaseHTTPServer.HTTPServer(
+                    ('localhost', port),
+                    SimpleHTTPServer.SimpleHTTPRequestHandler,
+                )
+            except socket.error:
+                time.sleep(0.5)
+
+    def _runserver(self):
+        self.httpd.serve_forever()
+
+    def makecert(self, key, cert, ip='127.0.0.1'):
+        subprocess.check_call([
+            'openssl',
+            'req', '-x509',
+            '-newkey', 'rsa:2048', '-sha256',
+            '-keyout', key,
+            '-out', cert,
+            '-days', '1',
+            '-nodes', '-subj',
+            '/CN={ip}'.format(ip=ip),
+        ])
+
+    def __enter__(self):
+        self.makecert(self.key, self.cert)
+        self.makecert(self.badkey, self.badcert, ip='127.0.0.2')
+        os.mkdir('sdk')
+        # We have to create the vimService file because the current version of
+        # pyvmomi seems happy to make an insecure request for the wsdl before
+        # it actually complains about SSL issues
+        with open('sdk/vimService.wsdl', 'w') as wsdl_handle:
+            wsdl_handle.write(
+                '<definitions xmlns="http://schemas.xmlsoap.org/wsdl/" '
+                'xmlns:soap="http://schemas.xmlsoap.org/wsdl/soap/" '
+                'xmlns:interface="urn:vim25" '
+                'targetNamespace="urn:vim25Service">\n'
+                '<import location="vim.wsdl" namespace="urn:vim25"/>\n'
+                '<service name="VimService">\n'
+                '<port binding="interface:VimBinding" name="VimPort">\n'
+                '<soap:address '
+                'location="https://localhost/sdk/vimService"/>\n'
+                '</port>\n'
+                '</service>\n'
+                '</definitions>\n'
+            )
+        self.process = multiprocessing.Process(
```

Comment:
Yeah, I think I'd then have to introduce select as I have a suspicion that the GIL would start to get involved in a noticeable way.